feat(sdk): Add `addGlobalEventProcessor` to Event Processors page (#5803)
getsentry/sentry-docs: src/platforms/common/enriching-events/event-processors.mdx

  - unreal
---

> With event processors, you can enrich events with additional data. You can add your own event processors to either the current scope or the global one. The difference between `beforeSend` and event processors is that `beforeSend` is guaranteed to be run last, after all other event processors. In addition, event processors added with `addEventProcessor` run on the scope level, whereas `beforeSend` (and event processors added with `addGlobalEventProcessor`) run globally, no matter which scope you're in. Like `beforeSend`, event processors are passed both the event itself and <PlatformLink to="/configuration/filtering/#using-hints">a `hint`` object</PlatformLink> containing extra metadata.
> 
> Event processors added to the global scope will run on every event sent after they are added.

Reviewer: ```suggestion
You can enrich an event with additional data by running `eventProcessors` on either the scope level or globally. The difference between `beforeSend` and `eventProcessors` is that `beforeSend` always runs last (after all other event processors), and globally. While `eventProcessors` added with `addEventProcessor` run on the scope level, and `eventProcessors` added with `addGlobalEventProcessor`, run globally on every event sent after they are added, no matter which scope you’re in. 

Both `beforeSend` and `eventProcessors` are passed the event itself and  [a `hint` object](https://sentry-docs-git-kmclb-add-addglobaleventprocessor.sentry.dev/platforms/javascript/configuration/filtering/#using-hints)  containing extra metadata.
```

Author: Yup, I was consolidating what was there and changing what I needed to more than really wordsmithing, so thanks for cleaning it up! A few things I'm going to push back on, though:

- The user adds the event processors, then the SDK runs them. So that word choice is deliberate (and matches the method names `addXXX`).
- Can "while" be used on its own like that? I think of it as something which can only introduce a subordinate clause.
- On the writing-event-processors-in-code question, I don't think it's correct to treat them as code. `eventProcessors` is essentially a proper noun, the name of a variable. 'Event processors' is either an adjective-improper-noun pair or a compound improper noun, and that's how we're using them, to talk about the behavior of the overall class of things.

I don't disagree that the third sentence could be simplified and broken up, though. Lemme give it another shot.

  - unreal
---

You can enrich events with additional data by adding your own event processors, either on the scope level or globally. Though event processors are similar to `beforeSend`, there are two key differences:

> - `beforeSend` is guaranteed to be run last, after all other event processors, which means it gets the final version of the event version of the event right before it's sent (hence the name!). Event processors added with either of the methods below run in an undetermined order, meaning changes to the event may still be made after the event processor runs.

Reviewer: ```suggestion
- `beforeSend` is guaranteed to be run last, after all other event processors, which means it gets the final version of the event right before it's sent (hence the name). Event processors added with either of the methods below run in an undetermined order, which means changes to the event may still be made after the event processor runs.
```

Author: Yikes. Proofreading. Thanks.

> - While `beforeSend` and processors added with `Sentry.addGlobalEventProcessor` run globally, regardless of scope, processors added with `scope.addEventProcessor` only run on events captured while the scope in question is active.
> 

Reviewer: ```suggestion
- While `beforeSend` and processors added with `Sentry.addGlobalEventProcessor` run globally, regardless of scope, processors added with `scope.addEventProcessor` only run on events captured while that scope is active.

```
